kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/URLFromServiceImplUtil.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.client.utils;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.api.model.extensions.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class URLFromServiceImplUtil {
+  public static Logger logger = LoggerFactory.getLogger(URLFromServiceImplUtil.class);
+  public static final String DEFAULT_PROTO = "tcp";
+  private static final String HOST_SUFFIX = "_SERVICE_HOST";
+  private static final String PORT_SUFFIX = "_SERVICE_PORT";
+  private static final String PROTO_SUFFIX = "_TCP_PROTO";
+
+  public static String serviceToHostOrBlank(String serviceName) {
```

Comment:
I will call it `resolveHostFromEnvVarOrSystemProperty` and the same for port, proto, .... because when I read the code that calls this method I was not sure if it were resolving the env varor not.